### DISCUSSION TITLE
Static Site Generator

### DISCUSSION THREAD

**Discussion by royemosby** [nditcommunity/ndit-website#49] [Technology]
Scaffolding out and managing multiple pages becomes a difficult and error prone endeavor over time. To minimize code duplication and layout inconsistencies between pages, we can turn page sections into re-usable templates(components) that can be stitched together through a build tool called a static site generator. There are plenty options out there thanks to the [JAM-Stack](https://jamstack.org/what-is-jamstack/) craze of a few years back. Lightweight, SEO friendly, and minimalistic are all desired characteristics for our website.

We still need to use a technology. For this, I am biased towards Eleventy but there are other tools that can be just as useful and these are the ones we are looking at:

- [Eleventy](https://www.11ty.dev/)
- [Astro](https://astro.build/)
- [Hugo](https://gohugo.io/)
- [Jekyll](https://jekyllrb.com/)

What are your favored tools and their pros and cons? I would like to get this closed out by the 24rd of April so that we can start a build.

**Comment by royemosby** [comment]
I am in favor of Eleventy because of its ease of use and the fact that it doesn't do anything with your code you don't ask for. I also like the tool for its variety of [templating and file formats](https://www.11ty.dev/docs/languages/) supported. I picked up Liquid while using Jekyll.

**Reply by marcusolsson** [reply]
I'd vote for Hugo or Eleventy. I don't have any personal experience with Eleventy, but I support their philosophy. I've been meaning to check it out, so I don't mind going with it for learning purposes, if you feel confident using it 👍

**Reply by marcusolsson** [reply]
I think we should prioritize moving to a SSG, since it'll impact other design decisions.

**Reply by hayleyw7** [reply]
I have no experience with any. I lean toward one that's common with lots of online help and likeliness of being solid future marketable experience on resumes. Ultimately I will 100% defer to anyone else who has real opinions on this.
